### ISSUE TITLE
Debug toolbar buttons disabled

### ISSUE THREAD

**Issue by testforstephen** [Microsoft/vscode#58327]
VSCode:  Version 1.27.1 (1.27.1) 5944e81f3c46a3938a82c701f96d7a59b074cfdc
Platform: macOS/ubuntu

When the user is stepping over his program, the CALL STACK view shows the program is PAUSED ON STEP, but the debug toolbar buttons are disabled. See the following screenshot.
<img width="1437" alt="screen shot 2018-09-10 at 9 20 53 am" src="https://user-images.githubusercontent.com/14052197/45271839-41877000-b4dc-11e8-8e34-88c5a7e8ddfb.png">

This issue is stably reproduced on macOS/ubuntu. Also there is some user reporting it occurs in Windows 10 sometimes, and downgrade to 1.26.0, the issue disappear.

**Comment by tanneraphid34** [comment]
![image](https://user-images.githubusercontent.com/30430070/45414138-5577d100-b698-11e8-8787-26c1b7b77b14.png)
The issue is coming in Win10 as well. 
i'm using Win10 Enterprise(Version 1607, build 14393.2430)

**Comment by Sindisil** [comment]
Seeing this when attempting to debug a Java program. Reverting to VSCode 1.26 works around the issue for me as well.

**Comment by testforstephen** [comment]
@isidorn  Any updates about this bug?

**Comment by isidorn** [comment]
@testforstephen I was on vacation, sorry for the slow response.
First of all can you reproduce this with the latest vscode insiders?

**Comment by Sindisil** [comment]
Not sure if @testforstephen can repro with latest vscode insiders, but I just updated my vscode insiders install, and it does indeed still repro.

**Comment by testforstephen** [comment]
@isidorn yes, this issue can be easily reproduced in 2018-09-17 windows insider version.

**Comment by weinand** [comment]
Any idea why this problem only affects Java debugging? We have no other reports of this.

**Comment by testforstephen** [comment]
@weinand In Java Debugger, we took asynchronous programming. One thread to handle the DA request,  another thread to report event from JVM in real time.  That means when the user is stepping, the possible DA sequence is `[ContinueRequest, New StoppedEvent, ContinueResponse]`, or `[ContinueRequest, ContinueResponse, New StoppedEvent]`. I guess this may be the reason.

**Comment by weinand** [comment]
Ok, I see.
DAP requires that the ContinueResponse is received before the resulting StoppedEvent.

@isidorn did we change anything in that area in 1.27.1?

**Comment by testforstephen** [comment]
The same logic works previously in a long time, we didn't change that part code recently. But i'll do some experiment on your suggestion.

**Comment by testforstephen** [comment]
through the screenshot we pasted above, apprently vscode knows the thread is paused on step, because the CALL STACK view/variable/code pointer are rendered correctly, except that the debug buttons don't respect the same status.

Besides, i did a quick test via keeping the stoppedEvent sent after StepResponse, it can mitigate the status issue for single thread. But the debug buttons still have the inconsistent status issue for multiple threads case.
For example, there are multiple threads suspending on breakpoints at the same time, continue one thread to make it's becoming running, then switch to the other thread, it's CALL STACK status/code pointer both are stopping at the breakpoint, but the step buttons are disabled. Just like the bug described.  Looks like this is a status management issue that should be investigated at vscode side.

**Comment by bastlaca** [comment]
I can also repeat this problem:
1. Start (tomcat) debuggint
2. Click step over(F10)
After that Continue(F5) will turn into Pause(F6) and central buttons will be grays.

I use:
- VS Code 1.27.2
- Linux Mint 19

**Comment by SGI495** [comment]
I (and multiple colleagues) can also reproduce this issue.

Same description: Debugger stops on breakpoints, I step over once and then the thread is paused on step but debug controls are deactivated.

Let me know if I can try anything or send any info to help nail down the issue.

* VSCode 1.27.2
* Windows 10

**Comment by testforstephen** [comment]
we submit a fix on java-debug side to fix the StoppedEvent sequence issue, now the debug buttons work well for the single thread scenario. But if it's multiple thread scenario, switching to other thread, VS Code probably lose the old thread's debug status. See the gif.
![debugbuttons](https://user-images.githubusercontent.com/14052197/45752037-cb56dc00-bc46-11e8-9b65-4e6ed5981315.gif)

**Comment by SGI495** [comment]
@testforstephen I just saw the comment in another issue about the debugger fix. I've updated to version 0.13 of the Debugger for Java and it works again for me now. Thanks!

**Comment by weinand** [comment]
@isidorn since this problem already occurred in 1.27 it is not related to my changes for 1.28

**Comment by bastlaca** [comment]
It works again. Thank you.

**Comment by testforstephen** [comment]
Verified the multiple threads case, it works in the 2018-09-20 windows insider.

**Comment by isidorn** [comment]
@testforstephen thanks a lot for verifiing, adding verified label
